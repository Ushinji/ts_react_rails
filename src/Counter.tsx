import * as React from 'react';

const Counter: React.FC = () => {
  const [count, SetCount] = React.useState(0);

  const handleClick = () => {
    SetCount(count + 1);
  };

  return (
    <div>
      <h1>カウンター</h1>
      <button type="button" onClick={handleClick}>
        ＋
      </button>
      <div>{count}</div>
    </div>
  );
};

export default Counter;
