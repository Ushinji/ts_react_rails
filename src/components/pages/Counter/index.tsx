import * as React from 'react';
import styled from 'styled-components';
import Button from '@/components/atoms/Button';
import useBool from '@/modules/useBool';

const Container = styled.div`
  background-color: white;
  box-shadow: 0px 0px 4px rgba(51, 23, 20, 0.2);
  padding: 16px;
  width: 340px;
  border-radius: 4px;
`;

const Title = styled.div`
  font-size: 24px;
  font-weight: bold;
  margin-bottom: 16px;
`;

const Counter: React.FC = () => {
  const [submitting, startSubmit, finishSubmit] = useBool(false);
  const [count, SetCount] = React.useState(0);

  const handleClick = () => {
    startSubmit();
    setTimeout(() => {
      SetCount(count + 1);
      finishSubmit();
    }, 1000);
  };

  return (
    <Container>
      <Title>
        カウンター：
        {count}
      </Title>
      <Button type="button" onClick={handleClick} submitting={submitting}>
        Count up
      </Button>
    </Container>
  );
};

export default Counter;
