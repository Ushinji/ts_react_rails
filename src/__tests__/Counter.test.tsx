import * as React from 'react';
import { render } from 'react-testing-library';
import Counter from '../Counter';

describe('Counter', () => {
  it('Snapshot test', () => {
    const { asFragment } = render(<Counter />);
    expect(asFragment()).toMatchSnapshot();
  });
});
