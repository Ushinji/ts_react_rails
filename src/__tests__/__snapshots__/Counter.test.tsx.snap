// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Snapshot test 1`] = `
<DocumentFragment>
  <div>
    <h1>
      カウンター
    </h1>
    <button
      type="button"
    >
      ＋
    </button>
    <div>
      0
    </div>
  </div>
</DocumentFragment>
`;
